stock-usecase/src/main/java/org/dhis2/android/rtsm/ui/home/screens/AnalyticsScreen.kt

```diff
@@ -0,0 +1,76 @@
+package org.dhis2.android.rtsm.ui.home.screens
+
+import android.widget.FrameLayout
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.material.BackdropScaffold
+import androidx.compose.material.BackdropValue
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.ScaffoldState
+import androidx.compose.material.rememberBackdropScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.core.view.ViewCompat
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.viewmodel.compose.viewModel
+import dhis2.org.analytics.charts.ui.GroupAnalyticsFragment
+import kotlinx.coroutines.CoroutineScope
+import org.dhis2.android.rtsm.ui.home.HomeViewModel
+import org.dhis2.android.rtsm.ui.home.screens.components.AnalyticsTopBar
+import org.dhis2.android.rtsm.ui.managestock.ManageStockViewModel
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun AnalyticsScreen(
+    viewModel: HomeViewModel = viewModel(),
+    manageStockViewModel: ManageStockViewModel = viewModel(),
+    themeColor: Color,
+    modifier: Modifier = Modifier,
+    scaffoldState: ScaffoldState,
+    syncAction: (scope: CoroutineScope, scaffoldState: ScaffoldState) -> Unit = { _, _ -> },
+    supportFragmentManager: FragmentManager,
+) {
+    val backdropState = rememberBackdropScaffoldState(BackdropValue.Revealed)
+    val settingsUiState by viewModel.settingsUiState.collectAsState()
+    val programName by viewModel.programDisplayName.collectAsState()
+    BackdropScaffold(
+        modifier = modifier,
+        appBar = {
+            AnalyticsTopBar(
+                title = programName,
+                themeColor = themeColor,
+                launchBottomSheet = {
+                    manageStockViewModel.onHandleBackNavigation()
```

Comment:
ManageStockVIewModel is for handling Dataentry, do we need to couple it with the analytics screen?

Reply:
You are correct, I can just pass the action on